### ISSUE TITLE
Building with a shared cargo dir?

### ISSUE THREAD

**Issue by skade** [habitat-sh/habitat#1129]
Hi,

I just tested building with a shared target dir for all subprojects against the current way of building and, due to sharing of compiled, dependencies, the build is much faster:

```
vagrant@vagrant:~/habitat$ time CARGO_TARGET_DIR=$PWD/target make
...
real	6m42.561s
user	6m12.956s
sys 0m18.380s
```

vs:

```
vagrant@vagrant:~/habitat$ time make
...
real	9m1.661s
user	8m20.124s
sys 0m26.660s
```

This also has the (convenient?) effect that all binaries end up in `$PWD/target/debug`.

Would a change for this be accepted or is there a reason why all projects use their own target dir?

If not, I'd be up to submitting a pull request.

**Comment by reset** [comment]
@skade thanks for the report! We definitely could gain some speed by sharing a common target directory. This should be solved by #1136 along with a few other pain points that we've had in our release process

**Comment by reset** [comment]
Closed by #1136
